pkg/provider/podsTracker_test.go

```diff
@@ -263,3 +263,81 @@ func TestCleanupDanglingPods(t *testing.T) {
 		}
 	}
 }
+
+func TestUpdatePodsLoop(t *testing.T) {
+	podName1 := "pod-" + uuid.New().String()
+	podName2 := "pod-" + uuid.New().String()
+	podName3 := "pod-" + uuid.New().String()
+	podNamespace := "ns-" + uuid.New().String()
+
+	podsNames := []string{podName1, podName2, podName3}
+	k8sPods := testsutil.CreatePodsList(podsNames, podNamespace)
+
+	k8sPods[0].Status.Phase = v1.PodPending
+	k8sPods[1].Status.Phase = v1.PodRunning
+	k8sPods[2].Status.Phase = v1.PodPending
```

Comment:
May we add the `PodSucceeded ` and `PodFailed ` status phases as well?

Reply:
There is a separate unit test for processPodUpdates that tests all the statuses, do we need that in the current unit test as well?

Reply:
Ah, you're right. But I can see the `PodFailed` is covered, but `PodSucceeded` is not. Feel free to add it to the other UT.